pkg/apis/resource/types.go

```diff
@@ -449,10 +450,21 @@ type DeviceRequest struct {
 	// +optional
 	// +featureGate=DRAAdminAccess
 	AdminAccess *bool
+
+	// FirstAvailableOf contains subrequests, exactly one of which must be satisfied
+	// in order to satisfy this request. This field may only be set in the
+	// entries of DeviceClaim.Requests. It must not be set in DeviceRequest
+	// instances that themselves are part of a FirstAvailableOf.
+	//
+	// +optional
+	// +oneOf=deviceRequestType
+	// +listType=atomic
```

Comment:
add +featureGate annotation

Reply:
Sometimes feature gates are implemented so that the code for handling the feature simply gets skipped. This has the advantage that a bug in the code also gets skipped, but it leads to a poor user experience.

For admin access we chose instead to make the scheduler reject handling claims with the field set when the feature gate is off. Let's do the same here.

Reply:
Yeah, I've added a similar check for this feature in the allocator as exists for the adminAccess feature: https://github.com/kubernetes/kubernetes/pull/128586/commits/7f2922b89033f5181141b32a99a46757123a2aed#diff-8907e59dbe7c433690f8025b6cc248893a60478f6e751d4679bf0e64aa4471e7R172.

But the handling of the new feature in the registry/resource/{resourceclaim,resourceclaimtemplate} code (https://github.com/kubernetes/kubernetes/pull/128586/commits/e943a3cff71dab5e83b2f8f3d16dd4426690f820) also seems relevant here and I'm a bit uncertain about my implementation there. I've looked at the code for the adminAccess, but the situation here seems a bit different.

Reply:
https://github.com/kubernetes/kubernetes/pull/128586#discussion_r1856918797 and https://github.com/kubernetes/kubernetes/pull/128586#discussion_r1856920145, right? Or was there something else where you were unsure?